### ISSUE TITLE
Internal Compiler Error when compiling GPT2

### ISSUE THREAD

**Issue by gnawpaul** [aws-neuron/aws-neuron-sdk#639]
I'm trying to compile a GPT2 model and got an error message. The model compiled successfully in torchscript. I believe that it's rooted in some broadcasting issue.

Here's what I ran:
```
from transformers import GPT2Tokenizer, GPT2Model
tokenizer = GPT2Tokenizer.from_pretrained(model_path)
tokenizer.pad_token = tokenizer.eos_token
model = GPT2Model.from_pretrained(model_path, torchscript=True, return_dict=False)
model.eval()

neuroncore_pipeline_cores = 4
sequence_size = 1024

inputs = tokenizer(["This is a test"], return_tensors="pt", max_length=sequence_size, padding='max_length')
neuron_pipeline_model = torch.neuron.trace(model,
                                           example_inputs = inputs['input_ids'],
                                           compiler_args = ['--neuroncore-pipeline-cores', str(neuroncore_pipeline_cores)],
                                          )
```

Error message:
```
INFO:Neuron:There are 2 ops of 1 different types in the TorchScript that are not compiled by neuron-cc: aten::embedding, (For more information see[ https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/compiler/neuron-cc/neuron-cc-ops/neuron-cc-ops-pytorch.html](https://awsdocs-neuron.readthedocs-hosted.com/en/latest/release-notes/compiler/neuron-cc/neuron-cc-ops/neuron-cc-ops-pytorch.html))
INFO:Neuron:Number of arithmetic operators (pre-compilation) before = 2444, fused = 2433, percent fused = 99.55%


WARNING:tensorflow:From /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/torch_neuron/ops/aten.py:2277: where (from tensorflow.python.ops.array_ops) is deprecated and will be removed in a future version.
Instructions for updating:
Use tf.where in 2.0, which has the same broadcast rule as np.where


INFO:Neuron:Compiler args type is <class 'list'> value is ['--neuroncore-pipeline-cores', '4']
INFO:Neuron:Compiling function _NeuronGraph$956 with neuron-cc
INFO:Neuron:Compiling with command line: '/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/bin/neuron-cc compile /tmp/tmpi916968s/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /tmp/tmpi916968s/graph_def.neff --io-config {"inputs": {"0:0": [[1, 1024], "int64"], "1:0": [[], "int64"], "2:0": [[1, 1024, 1024], "float32"], "3:0": [[1, 1024, 1024], "float32"]}, "outputs": ["aten_view/Reshape:0", "GPT2Block_90/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_90/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_2/transpose:0"]} --neuroncore-pipeline-cores 4 --verbose 35'


...
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: ***************************************************************
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:  An Internal Compiler Error has occurred
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: ***************************************************************
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: 
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Error message:  Traceback (most recent call last):
  [bt] (8) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x8edf26) [0x7f1c32443f26]
  [bt] (7) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x8f2392) [0x7f1c32448392]
  [bt] (6) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x8e8bd1) [0x7f1c3243ebd1]
  [bt] (5) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x8e673b) [0x7f1c3243c73b]
  [bt] (4) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x8ebf11) [0x7f1c32441f11]
  [bt] (3) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x9afb50) [0x7f1c32505b50]
  [bt] (2) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0xaa6840) [0x7f1c325fc840]
  [bt] (1) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0xac629b) [0x7f1c3261c29b]
  [bt] (0) /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/libtvm.so(+0x5edf75) [0x7f1c32143f75]
  File "/opt/brazil-pkg-cache/packages/DmlcTvm/DmlcTvm-1.13.0.0/AL2_x86_64/generic-flavor/src/topi/include/topi/broadcast.h", line 32
TVMError: Check failed: output_shape.size() >= t->shape.size() (0 vs. 1) : Not a broadcast, output dimensionality smaller than input.
output: []
vs
input: Tensor(shape=[1], op.name=dram://placeholder15)
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: 
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Error class:    TVMError
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Error location: Unknown
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Command line:   /home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/bin/neuron-cc compile /tmp/tmpi916968s/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /tmp/tmpi916968s/graph_def.neff --io-config '{"inputs": {"0:0": [[1, 1024], "int64"], "1:0": [[], "int64"], "2:0": [[1, 1024, 1024], "float32"], "3:0": [[1, 1024, 1024], "float32"]}, "outputs": ["aten_view/Reshape:0", "GPT2Block_90/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_90/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_2/transpose:0"]}' --neuroncore-pipeline-cores 4 --verbose 35
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: 
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Internal details:
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/CommandDriver.py", line 224, in neuroncc.driver.CommandDriver.CommandDriver.run
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 576, in neuroncc.driver.commands.CompileCommand.CompileCommand.run
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 554, in neuroncc.driver.commands.CompileCommand.CompileCommand.runPipeline
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/commands/CompileCommand.py", line 558, in neuroncc.driver.commands.CompileCommand.CompileCommand.runPipeline
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/Pipeline.py", line 30, in neuroncc.driver.Pipeline.Pipeline.runSingleInput
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/Pipeline.py", line 30, in neuroncc.driver.Pipeline.Pipeline.runSingleInput
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/Job.py", line 289, in neuroncc.driver.Job.SingleInputJob.run
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 427, in neuroncc.driver.jobs.Frontend.Frontend.runSingleInput
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 377, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 378, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "neuroncc/driver/jobs/Frontend.py", line 382, in neuroncc.driver.jobs.Frontend.Frontend.runTVMFrontend
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/relay/build_module.py", line 731, in build_graph
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:     func = pre_partitioning_passes(func, params)
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/relay/build_module.py", line 282, in pre_partitioning_passes
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:     func = ir_pass.fold_constant(func)
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/relay/ir_pass.py", line 707, in fold_constant
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:     return _ir_pass.FoldConstant(expr)
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/tvm/_ffi/_ctypes/function.py", line 190, in __call__
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:     raise get_last_ffi_error()
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: 
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Version information:
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   Neuron Compiler version 1.13.5.0+7dcf000a6
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   HWM version 1.13.0.0-0
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   NEFF version Dynamic
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   TVM version 1.13.0.0+0
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   NumPy version 1.18.5
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   MXNet not available
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]:   TF not available
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: 
03/17/2023 12:00:36 AM ERROR 12345 [neuron-cc]: Artifacts stored in: /tmp/tmpi916968s


Compiler status ERROR


INFO:Neuron:Compile command returned: 1
WARNING:Neuron:torch.neuron.trace failed on _NeuronGraph$956; falling back to native python function call
ERROR:Neuron:neuron-cc failed with the following command line call:
/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/bin/neuron-cc compile /tmp/tmpi916968s/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /tmp/tmpi916968s/graph_def.neff --io-config '{"inputs": {"0:0": [[1, 1024], "int64"], "1:0": [[], "int64"], "2:0": [[1, 1024, 1024], "float32"], "3:0": [[1, 1024, 1024], "float32"]}, "outputs": ["aten_view/Reshape:0", "GPT2Block_90/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_90/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_2/transpose:0"]}' --neuroncore-pipeline-cores 4 --verbose 35
Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/torch_neuron/convert.py", line 392, in op_converter
    item, inputs, compiler_workdir=sg_workdir, **kwargs)
  File "/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/torch_neuron/decorators.py", line 229, in trace
    'neuron-cc failed with the following command line call:\n{}'.format(command))
subprocess.SubprocessError: neuron-cc failed with the following command line call:
/home/ec2-user/anaconda3/envs/amazonei_pytorch_latest_p37/bin/neuron-cc compile /tmp/tmpi916968s/graph_def.pb --framework TENSORFLOW --pipeline compile SaveTemps --output /tmp/tmpi916968s/graph_def.neff --io-config '{"inputs": {"0:0": [[1, 1024], "int64"], "1:0": [[], "int64"], "2:0": [[1, 1024, 1024], "float32"], "3:0": [[1, 1024, 1024], "float32"]}, "outputs": ["aten_view/Reshape:0", "GPT2Block_90/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_90/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_92/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_94/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_96/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_98/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_100/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_102/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_104/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_106/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_108/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_110/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_112/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_114/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_116/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_118/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_120/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_122/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_124/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_126/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_128/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_130/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_132/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_134/GPT2Attention_5/aten_permute_2/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_1/transpose:0", "GPT2Block_136/GPT2Attention_5/aten_permute_2/transpose:0"]}' --neuroncore-pipeline-cores 4 --verbose 35
INFO:Neuron:Number of arithmetic operators (post-compilation) before = 2444, compiled = 0, percent compiled = 0.0%
INFO:Neuron:The neuron partitioner created 1 sub-graphs
INFO:Neuron:Neuron successfully compiled 0 sub-graphs, Total fused subgraphs = 1, Percent of model sub-graphs successfully compiled = 0.0%
INFO:Neuron:Compiled these operators (and operator counts) to Neuron:
INFO:Neuron:Not compiled operators (and operator counts) to Neuron:
INFO:Neuron: => aten::Int: 557 [supported]
INFO:Neuron: => aten::ScalarImplicit: 25 [supported]
INFO:Neuron: => aten::add: 98 [supported]
INFO:Neuron: => aten::addmm: 96 [supported]
INFO:Neuron: => aten::arange: 1 [supported]
INFO:Neuron: => aten::contiguous: 24 [supported]
INFO:Neuron: => aten::div: 24 [supported]
INFO:Neuron: => aten::dropout: 73 [supported]
INFO:Neuron: => aten::embedding: 2 [not supported]
INFO:Neuron: => aten::full: 48 [supported]
INFO:Neuron: => aten::layer_norm: 49 [supported]
INFO:Neuron: => aten::matmul: 48 [supported]
INFO:Neuron: => aten::mul: 96 [supported]
INFO:Neuron: => aten::permute: 96 [supported]
INFO:Neuron: => aten::pow: 48 [supported]
INFO:Neuron: => aten::size: 555 [supported]
INFO:Neuron: => aten::slice: 96 [supported]
INFO:Neuron: => aten::softmax: 24 [supported]
INFO:Neuron: => aten::split: 24 [supported]
INFO:Neuron: => aten::sub: 24 [supported]
INFO:Neuron: => aten::tanh: 24 [supported]
INFO:Neuron: => aten::to: 72 [supported]
INFO:Neuron: => aten::transpose: 24 [supported]
INFO:Neuron: => aten::unsqueeze: 1 [supported]
INFO:Neuron: => aten::view: 291 [supported]
INFO:Neuron: => aten::where: 24 [not supported]


---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<timed exec> in <module>

~/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/torch_neuron/convert.py in trace(func, example_inputs, fallback, op_whitelist, minimum_segment_size, subgraph_builder_function, subgraph_inputs_pruning, skip_compiler, debug_must_trace, allow_no_ops_on_neuron, compiler_workdir, dynamic_batch_size, compiler_timeout, single_fusion_ratio_threshold, _neuron_trace, compiler_args, optimizations, verbose, **kwargs)
    193         logger.debug("skip_inference_context - trace with fallback at {}".format(get_file_and_line()))
    194         neuron_graph = cu.compile_fused_operators(neuron_graph, **compile_kwargs)
--> 195     cu.stats_post_compiler(neuron_graph)
    196 
    197     # Wrap the compiled version of the model in a script module. Note that this is

~/anaconda3/envs/amazonei_pytorch_latest_p37/lib/python3.7/site-packages/torch_neuron/convert.py in stats_post_compiler(self, neuron_graph)
    501         if succesful_compilations == 0 and not self.allow_no_ops_on_neuron:
    502             raise RuntimeError(
--> 503                 "No operations were successfully partitioned and compiled to neuron for this model - aborting trace!")
    504 
    505         if percent_operations_compiled < 50.0:

RuntimeError: No operations were successfully partitioned and compiled to neuron for this model - aborting trace!
```

Based on the broadcasting error message and some experiments, I narrowed it down to several lines of code in `transformers.gpt2.modeling_gpt2.GPT2Attention._attn` where we try to divide by some broadcasted tensor:
```
if self.scale_attn_weights:
    attn_weights = attn_weights / torch.full([], value.size(-1) ** 0.5, dtype=attn_weights.dtype, device=attn_weights.device
```

Changing the divisor into a scalar or vector using `torch.full_like` seemed to work.
A minimal reproducible example in my environment:
```
def test_divide(inputs):
    inputs = inputs / torch.full(
                [], 2, dtype=inputs.dtype, device=inputs.device
            )
    return inputs
neuron_pipeline_model = torch.neuron.trace(test_divide,
                                           example_inputs = torch.ones(2,2,512),
                                           verbose="debug",
                                          )
```
Tracing with torchscript worked fine:
```
traced_model = torch.jit.trace(test_divide, torch.ones(2,2,512))
```

My environment is a sagemaker notebook instance:
```
aws-neuron-dkms.noarch                                  2.2.8.0-dkms                   @neuron    
aws-neuron-runtime-base.x86_64                   1.6.21.0-1                     @neuron    
aws-neuron-tools.x86_64                                  2.0.790.0-1                    @neuron    
tensorflow-model-server-neuronx.x86_64      2.8.0.2.5.6.0-0                @neuron  
```

```
torch                         1.12.1
torch-neuron           1.12.1.2.5.8.0
torcheia                    1.0.0
torchvision                0.13.1
neuron-cc                 1.13.5.0+7dcf000a6
```

**Comment by aws-donkrets** [comment]
Sorry you are seeing a failure - thanks for submitting the ticket.  Your effort to narrow it down to a change in the divisor may suggest an issue in the underlying TVM library. We are investigating the issue.

**Comment by aws-taylor** [comment]
Hello @gnawpaul,

  We have identified the issue and found a fix. We will update this issue once the fix is present in a near release.

-T

**Comment by awsilya** [comment]
@gnawpaul the fix is available in newly released Neuron 2.10.  Please try it.

**Comment by gnawpaul** [comment]
We tried it and can confirm that it works for us. Thanks!
